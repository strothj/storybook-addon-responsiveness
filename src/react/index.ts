export { default as withResponsiveness } from "./withResponsiveness";
