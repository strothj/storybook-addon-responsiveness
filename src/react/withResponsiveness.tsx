import { StoryDecorator } from "@storybook/react";
import addons from "../addons";
import { eventEnableStoryResponsiveness } from "../constants";

const withResponsiveness: StoryDecorator = (render, { kind, story }) => {
  const channel = addons.getChannel();

  channel.emit(eventEnableStoryResponsiveness, { kind, story });

  return render() as any;
};

export default withResponsiveness;
