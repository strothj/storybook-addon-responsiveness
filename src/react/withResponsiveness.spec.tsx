import * as React from "react";
import withResponsiveness from "./withResponsiveness";

const mockEmit = jest.fn();

jest.mock("../addons", () => {
  const addonsModule = { getChannel: () => ({ emit: mockEmit }) };
  return { default: addonsModule };
});

beforeEach(() => {
  mockEmit.mockReset();
});

it("emits event with story kind and name", () => {
  withResponsiveness(() => <div>Component</div>, {
    kind: "kind",
    story: "story",
  });

  expect(mockEmit.mock.calls).toMatchSnapshot();
});

it("renders component", () => {
  const render = withResponsiveness(() => <div>Component</div>, {
    kind: "kind",
    story: "story",
  });

  expect(render).toMatchSnapshot();
});
