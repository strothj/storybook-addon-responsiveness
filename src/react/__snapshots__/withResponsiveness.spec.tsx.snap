// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emits event with story kind and name 1`] = `
Array [
  Array [
    "strothj/storybook-addon-responsiveness/enable_story_responsiveness",
    Object {
      "kind": "kind",
      "story": "story",
    },
  ],
]
`;

exports[`renders component 1`] = `
<div>
  Component
</div>
`;
